### PR TITLE
ts: upgrade anchor

### PR DESCRIPTION
**Pull request by microwavedcola1**

### CODE REVIEW

**Review by ckamm — APPROVED** [review]

File: ts/client/src/client.ts [hunk 1]
```diff
@@ -1855,11 +1855,11 @@ export class MangoClient {
     group: Group,
     mangoAccount: MangoAccount,
     externalMarketPk: PublicKey,
-    side: Serum3Side,
+    side: Serum3Side | any,
```
Comment (microwavedcola1) [inline ts/client/src/client.ts:1858]: image

So basically we would have to move away from current to globals, see both current and global example below
image

Comment (microwavedcola1) [reply ts/client/src/client.ts:1858]: I think my current workaround is better, we keep code as is, but have a `| any` so the caller knows the main type he should use, and code compiles

Comment (microwavedcola1) [reply ts/client/src/client.ts:1858]: Alternatively we could disable compile checking completely for these lines, or do you have other suggestions?

File: ts/client/src/accounts/mangoAccount.ts [hunk 2]
```diff
@@ -1793,15 +1793,29 @@ export class PerpOoDto {
   ) {}
 }
 
-export class TokenConditionalSwapDisplayPriceStyle {
-  static sellTokenPerBuyToken = { sellTokenPerBuyToken: {} };
-  static buyTokenPerSellToken = { buyTokenPerSellToken: {} };
+export type TokenConditionalSwapDisplayPriceStyle =
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | { sellTokenPerBuyToken: {} }
```
Comment (ckamm) [inline ts/client/src/accounts/mangoAccount.ts:1798]: I suggest using the recommended `Record<string, none>` over `{}` here so we can get rid of the lint-comment

**Review by ckamm — APPROVED** [review]

**Pull request merged**